packages/auth/src/authImplementations/netlify.ts

```diff
@@ -0,0 +1,120 @@
+import type * as NetlifyIdentityNS from 'netlify-identity-widget'
+
+import { isBrowser } from '@redwoodjs/prerender/browserUtils'
+
+import { CurrentUser } from 'src/AuthContext'
+import { createAuthentication } from 'src/authFactory'
+// TODO:
+// In the future, when this is a separate package, we can import the full thing
+// here (not just the type), and save the user from doing that in their own
+// code
+// import netlifyIdentity from 'netlify-identity-widget'
+
+import { AuthImplementation } from './AuthImplementation'
+
+type NetlifyIdentity = typeof NetlifyIdentityNS
+
+export function createNetlifyAuth(
+  netlifyIdentity: NetlifyIdentity,
+  customProviderHooks?: {
+    useCurrentUser?: () => Promise<Record<string, unknown>>
+    useHasRole?: (
+      currentUser: CurrentUser | null
+    ) => (rolesToCheck: string | string[]) => boolean
+  }
```

Comment:
@peterp I haven't converted auth0 to this new way of doing things yet, but here's how I imagine it could be possible to pass in a custom implementation of `getCurrentUser` for Netlify, and it would be very similar for all other auth service providers too.

In your app, when you setup auth you call `createNetlifyAuth` (or `createAuth0Auth` in your case). I added this `customProviderHooks` object to the create function that lets you pass in your custom implementations of `getCurrentUser` and `hasRole`. (Those were the only two that I thought made sense to expose, but I could totally add more if anyone sees any use for it).

Do you think this would work for your use case?

Also not entirely sure about the return type of the function that `useCurrentUser` returns. Maybe it could be more specific. Or more generic, and you pass in the type when you call `createNetlifyAuth`, like `createNetlifyAuth<CurrentUser>()`.

If you look at how I use these custom implementations in `AuthProvider.tsx` you can see I had to do `eslint-disable-next-line react-hooks/rules-of-hooks`. I wrote a comment about my reasoning on why that's safe. Would be awesome if you could have a look and see if you agree. Or, even better, if you have a better implementation idea 🙂

Reply:
> Do you think this would work for your use case?

Yup! I am expected to return exactly what `getCurrentUser` would return. As far as I know this has no bearing on the auth library used.